src/button/index.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const buttonOf = (Type) => {
+  const component = (props) => (
+    <Type {...props} className={`button${props.className ? ' ' + props.className : ''}`}>{props.children}</Type>
+  )
+  component.defaultName = "ButtonOfType" + Type
+  component.propTypes = {
+    children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+    onClick: PropTypes.func,
+    disabled: PropTypes.bool,
+    className: PropTypes.string
+  }
+  return component;
+}
+
+export const Button = buttonOf("button");
+Button.propTypes = Object.assign({ type: PropTypes.string }, Button.propTypes);
+
+export const LinkButton = buttonOf("a");
+LinkButton.propTypes = Object.assign({ href: PropTypes.string.isRequired }, LinkButton.propTypes);
+
+export default Button;
```

Comment:
I really like this, but the mixed use of double and singe quotes makes me twitch a little. I know it's wholly cosmetic, but could we standardise on single-quotes?

Reply:
Should be fixed now! This is why I like tools like prettier!